src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3307,18 +3307,23 @@ ntp_archiver::get_housekeeping_jobs() {
     return res;
 }
 
-ss::future<std::pair<
-  std::optional<ssx::semaphore_units>,
-  std::optional<upload_candidate_with_locks>>>
+ss::future<ntp_archiver::find_reupload_candidate_result>
 ntp_archiver::find_reupload_candidate(manifest_scanner_t scanner) {
     ss::gate::holder holder(_gate);
+    archival_stm_fence rw_fence{
+      .read_write_fence
+      = _parent.archival_meta_stm()->manifest().get_applied_offset(),
+      .unsafe_add = false,
```

Comment:
shouldn't `unsafe_add = false` be the default? i think we should only see this field being set in tests?

Reply:
> these tests

this comment isn't in a test

Reply:
my bad
I think you're right but the `unsafe_add` was renamed in the next commit and the default I think is also different. in any case it's a minor thing, can change in a followup